Add doc about extension backend mounting docker socket (#153)
docker/extensions-sdk: docs/extensions/METADATA.md

},
```

The vm metadata section should define either `image` or `composefile`. When you use `image`, a default compose file is generated for the extension.

> `${DESKTOP_PLUGIN_IMAGE}` is a specific keyword that allows an easy way to refer to the image packaging the extension. It is also possible to specify any other full image name here. However, in many cases using the same image makes things easier for extension development.

> ### Defining your own compose file for extension backend

Reviewer: ```suggestion
### Define your own compose file for the extension backend
```

},
```

> The composefile would then look like (for exampe with a volume definition):

Reviewer: ```suggestion
The composefile, with a volume definition for example, would look like:
```

```yaml
services:
  myExtension:
    image: ${DESKTOP_PLUGIN_IMAGE}
    volumes:
      - /host/path:/container/path
```

> ### Using the docker socket from your extension backend

Reviewer: ```suggestion
### Use the docker socket from your extension backend
```

> Docker extensions can invoke Docker commands directly from the UI js API. In some cases it can be useful to also interact with the Docker engine from the backend. Extension backend containers can mount the docker socket and use it to interact with the Docker engine from the extension backend logic.

Reviewer 2: ```suggestion
Docker extensions can invoke Docker commands directly from the frontend with the SDK. In some cases, it is useful to also interact with the Docker engine from the backend. Extension backend containers can mount the docker socket and use it to interact with the Docker engine from the extension backend logic.
```

> Docker extensions can invoke Docker commands directly from the UI js API. In some cases it can be useful to also interact with the Docker engine from the backend. Extension backend containers can mount the docker socket and use it to interact with the Docker engine from the extension backend logic.

Reviewer 2: Shouldn't we explain what the docker socket is for developers that don't know how docker engine and docker desktop internals work?

Author: I would assume developers who don't know what it is will likely not need this level of information, and need to learn more about Docker in general in order to develop complex extensions. (or they would just execute the docker CLI from the SDK for simple use cases)
I'm not sure if we should point them at some general Docker tutorial... I would assume if you read this you already know about the general Docker mechanisms

However, when mounting the docker socket from an extension container that lives in the Desktop virtual machine, you want to mount the Docker socket from inside the VM, and not mount `/var/run/docker.sock` from the host filsystem (using the Docker socket from the host can lead to permission issues in containers).

> In order to do so, you can use `/var/run/docker.sock.raw` and Docker Desktop will treat this specifically to mount the socket that lives in the Desktop VM, and not from the host.

Reviewer: Not sure about this sentence. I've added a suggestion, but not sure if this then changes the meaning of the sentence?
```suggestion
In order to do so, you can use `/var/run/docker.sock.raw`. Docker Desktop mounts the socket that lives in the Desktop VM, and not from the host.
```
